packages/yoroi-extension/app/components/wallet/assets/NFTsList.js

```diff
@@ -109,26 +168,80 @@ function NfTsList({ list, intl }: Props & Intl): Node {
 
 export default (injectIntl(NfTsList): ComponentType<Props>);
 
-function NftCardImage({ image, name }) {
-  let nftImage
-  if (image != null) {
-    const ipfsHash = image.replace('ipfs://', '');
-    nftImage = (<img src={`https://ipfs.io/ipfs/${ipfsHash}`} alt={name} loading="lazy" />);
-  } else {
-    nftImage = (<DefaultNFT />);
-  }
+function isImageExist(imageSrc: string, onload: void => void, onerror: void => void) {
+  const img = new Image();
+  img.onload = onload;
+  img.onerror = onerror;
+  img.src = imageSrc;
+}
+
+export function NftImage({ imageUrl, name, width, height }: {|
+  imageUrl: string | null,
+  name: string,
+  width: string,
+  height: string
+|}): Node {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  imageUrl = imageUrl !== null ? imageUrl.replace('ipfs://', 'https://ipfs.io/ipfs/'): null;
```

Comment:
Reasigning a prop is not a good practice, better create a new variable.

Reply:
@nistadev  Good point, Thanks!